app/src/main/java/com/arctouch/codechallenge/api/TmdbApi.java

```diff
@@ -36,4 +35,12 @@
             @Query("api_key") String apiKey,
             @Query("language") String language
     );
+
+    @GET("search/movie")
+    Observable<UpcomingMoviesResponse> searchMovies(
+            @Query("api_key") String apiKey,
+            @Query("query") String query,
```

Comment:
You have `apiKey` and `query` on each of your HTTP requests. Can you think of a way to not inform them on each request?

Reply:
Yes, but actually only api_key is repeated on every request, the query parameter is used only on searchMovies method, I could avoid repetition creating an interceptor on httpclient in APIHelper.java file. Like the following:
```
api = new Retrofit.Builder()
                    .baseUrl(TmdbApi.URL)
                    .client(new OkHttpClient.Builder().addInterceptor(new Interceptor() {
                        @Override
                        public Response intercept(Chain chain) throws IOException {
                            Request originalRequest = chain.request();
                            HttpUrl originalHttpUrl = originalRequest.url();

                            HttpUrl url = originalHttpUrl.newBuilder()
                                    .addQueryParameter("api_key", TmdbApi.API_KEY)
                                    .build();

                            Request.Builder requestBuilder = originalRequest.newBuilder()
                                    .url(url);

                            Request request = requestBuilder.build();
                            return chain.proceed(request);
                        }
                    }).build())
                    .addConverterFactory(MoshiConverterFactory.create())
                    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                    .build()
                    .create(TmdbApi.class);
```